sphinx/builders/linkcheck.py

```diff
@@ -378,6 +380,11 @@ def allowed_redirect(url: str, new_url: str) -> bool:
 
         def check(docname: str) -> Tuple[str, str, int]:
             # check for various conditions without bothering the network
+
+            for doc_matcher in self.documents_exclude:
+                if doc_matcher.match(docname):
+                    return 'unchecked', '', 0
```

Comment:
I think `'ignored'` would be better than `'unchecked'`. Unchecked is used for unsupported protocols. Ignored means linkcheck has been configured to ignore that link.

Also, can add a reason explaining why the link was ignored:
```suggestion
                    return 'ignored', f'{docname} matched {doc_matcher} from linkcheck_exclude_documents', 0
```